crates/runix/src/store_path.rs

```diff
@@ -0,0 +1,264 @@
+use std::ffi::OsStr;
+use std::fmt::Display;
+use std::path::{Component, Path, PathBuf};
+use std::str::FromStr;
+
+use once_cell::sync::Lazy;
+use thiserror::Error;
+
+/// Respect [NIX_STORE_DIR](https://nixos.org/manual/nix/stable/command-ref/env-common.html#env-NIX_STORE_DIR)
+pub static STORE_PREFIX: Lazy<String> = Lazy::new(|| {
+    std::env::var("NIX_STORE_DIR")
+        .ok()
+        .or_else(|| option_env!("NIX_STORE_DIR").map(String::from))
+        .unwrap_or_else(|| "/nix/store".to_string())
+});
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct StorePath {
+    prefix: PathBuf,
+    name: String,
+    package_path: Option<PathBuf>,
+}
+
+impl StorePath {
+    /// nix store prefix, checked against [STORE_PREFIX]
+    ///
+    /// ```
+    /// # use std::path::Path;
+    /// # use runix::store_path::{StorePath, STORE_PREFIX};
+    ///
+    /// let path = StorePath::from_path("/nix/store/7rjqb838snvvxcmpvck1smfxhkwzqal5-python3-3.10.10")
+    ///     .unwrap();
+    /// assert_eq!(path.prefix(), Path::new(STORE_PREFIX.as_str()));
+    /// ```
+    pub fn prefix(&self) -> &Path {
+        &self.prefix
+    }
```

Comment:
This seems kind of pointless. If people need it can't they just use STORE_PREFIX? Or if that's too hard to find, just return STORE_PREFIX here? Just seems like `Path::new(STORE_PREFIX.as_str()).to_path_buf(),` is wasteful

Reply:
I'd kinda prefer the object to be self contained rather than relying on static data outside.
I agree its not quite necessary right now, since we don't serialize them or do anything else to share those paths across different prefixes.
However i dont think a single string (though wasteful indeed right now) would weigh in that much to consider optimizing for it now?

Reply:
Not worth blocking on